File: java/src/test/java/ai/buble/sdk/FilesTest.java
package ai.buble.sdk;

import ai.buble.sdk.files.FileUpload;
import ai.buble.sdk.files.UploadOptions;
import org.junit.jupiter.api.Test;

import java.nio.charset.StandardCharsets;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

final class FilesTest {
    @Test
    void uploadsMultipartFile() throws Exception {
        try (TestHttpServer server = new TestHttpServer(request ->
                new TestHttpServer.Response(200, "{\"data\":{\"object\":\"file\",\"url\":\"https://example.com/reference.png\",\"key\":\"k\",\"file_type\":\"image\",\"content_type\":\"image/png\",\"size\":3,\"filename\":\"reference.png\"}}"))) {
            BubleClient client = BubleClient.builder().apiKey("sk_test").baseUrl(server.url()).build();

            client.files().upload(
                    FileUpload.fromBytes("abc".getBytes(StandardCharsets.UTF_8), "reference.png"),
                    UploadOptions.builder()
                            .fileType("image")
                            .model("google/nano-banana")
                            .mode("image_to_image")
                            .build());

            TestHttpServer.CapturedRequest request = server.requests().get(0);
            assertEquals("/api/v1/files", request.path);
            assertTrue(request.contentType.startsWith("multipart/form-data; boundary="));
            assertTrue(request.body.contains("name=\"file_type\""));
            assertTrue(request.body.contains("google/nano-banana"));
            assertTrue(request.body.contains("filename=\"reference.png\""));
            assertTrue(request.body.contains("abc"));
        }
    }
}

File: flutter/test/files_test.dart
import 'dart:convert';

import 'package:buble/buble.dart';
import 'package:test/test.dart';

import 'test_http_client.dart';

void main() {
  test('uploads multipart file', () async {
    final http = TestHttpClient([
      jsonResponse({
        'data': {
          'object': 'file',
          'provider': 'r2',
          'url': 'https://example.com/source.png',
          'key': 'api/image/source.png',
          'file_type': 'image',
          'content_type': 'image/png',
          'size': 3,
          'filename': 'source.png',
        },
      }),
    ]);
    final client = BubleClient(apiKey: 'sk_test', httpClient: http);

    final uploaded = await client.files.upload(
      FileUpload.fromBytes(
        [1, 2, 3],
        filename: 'source.png',
        contentType: 'image/png',
      ),
      options: const UploadOptions(
        fileType: 'image',
        model: 'google/nano-banana',
        mode: 'image_to_image',
      ),
    );

    final body = utf8.decode(http.requests.single.body, allowMalformed: true);
    expect(uploaded.data.url.toString(), 'https://example.com/source.png');
    expect(body, contains('name="file_type"'));
    expect(body, contains('image_to_image'));
    expect(body, contains('filename="source.png"'));
  });
}
